app/src/main/java/com/nowak/wjw/simplecompass/ui/main/MainViewModel.java

```diff
@@ -1,32 +1,37 @@
 package com.nowak.wjw.simplecompass.ui.main;
 
-import android.hardware.Sensor;
-import android.hardware.SensorEvent;
-import android.hardware.SensorManager;
-import android.location.Location;
-
 import androidx.lifecycle.LiveData;
+import androidx.lifecycle.LiveDataReactiveStreams;
 import androidx.lifecycle.MutableLiveData;
 import androidx.lifecycle.Transformations;
 import androidx.lifecycle.ViewModel;
 
-import com.nowak.wjw.simplecompass.CompassStateEnum;
 import com.nowak.wjw.simplecompass.Event;
 import com.nowak.wjw.simplecompass.R;
-
+import com.nowak.wjw.simplecompass.data.LocationCoordinates;
+import com.nowak.wjw.simplecompass.domain.GetAzimuthUseCase;
+import com.nowak.wjw.simplecompass.domain.GetDestinationBearingUseCase;
+import com.nowak.wjw.simplecompass.domain.InitiateLastLocationUseCase;
+import com.nowak.wjw.simplecompass.domain.RequestAndStopLocationUpdatesUseCase;
+import com.nowak.wjw.simplecompass.domain.StartStopSensorListenerUseCase;
+
+import io.reactivex.rxjava3.core.BackpressureStrategy;
 import timber.log.Timber;
 
 public class MainViewModel extends ViewModel {
-    private MutableLiveData<Integer> mAzimuth = new MutableLiveData<>(0);
-    private MutableLiveData<Location> mLocation = new MutableLiveData<>();
-    private MutableLiveData<Location> mDestination = new MutableLiveData<>();
-    private MutableLiveData<Event<Boolean>> mFoundLastLocation = new MutableLiveData<>(new Event<>(false));
-    private LiveData<Float> mDestinationBearing;
+    private StartStopSensorListenerUseCase mStartStopSensorListenerUseCase;
+    private InitiateLastLocationUseCase mInitiateLastLocationUseCase;
+    private RequestAndStopLocationUpdatesUseCase mRequestAndStopLocationUpdatesUseCase;
     private MutableLiveData<CompassStateEnum> mState = new MutableLiveData<>(CompassStateEnum.COMPASS_ONLY);
-    private MutableLiveData<Boolean> mStopLocationUpdates = new MutableLiveData<>(false);
-    private MutableLiveData<Boolean> mStartLocationUpdates = new MutableLiveData<>(false);
+    private MutableLiveData<Integer> mScreenOrientation = new MutableLiveData<>(0);
+    private LiveData<Integer> mAzimuth;
+    private MutableLiveData<LocationCoordinates> mDestinationCoordinates = new MutableLiveData<>();
```

Comment:
generalnie mam wrażenie że tego nie wiesz ale możę się mylę: mutableLiveData jest ukrywana po to żeby nikt z zewnątrz jej nie modyfikował - wystawiamy liveData jako immutableData które można obserwować. PRzez to że wszędzie używasz dataBinding używasz liveData co jest w sumie spoko.

Reply:
intencjonalnie pola MutableLiveData są prywatne, a wszystko co udostępniam, udostępniam jako LiveDate - właśnie, żeby nie można było przy tym grzebać.